### PR TITLE
F-Coalgebras form a category

### PR DESCRIPTION
**Pull request by Taneb**
More or less copied from the F-Algebras category. I wanted this while I was messing around with recursion schemes a while back.

### CODE REVIEW

File: src/Categories/Functor/Coalgebra.agda [hunk 1]
```diff
@@ -19,3 +19,12 @@ open F-Coalgebra
 -- Given a F-Coalgebra F, one can apply F to it to obtain an new 'iterated' F-Coalgebra
 iterate : ∀ {o ℓ e} {C : Category o ℓ e} {F : Endofunctor C} → F-Coalgebra F → F-Coalgebra F
 iterate {F = F} F-alg = record { A = Functor.F₀ F $ A F-alg ; α = Functor.F₁ F $ α F-alg }
+
+record F-Coalgebra-Morphism {o ℓ e} {C : Category o ℓ e} {F : Endofunctor C} (X Y : F-Coalgebra F) : Set (ℓ ⊔ e) where
+  open Category C
+  module X = F-Coalgebra X
```
Comment (JacquesCarette) [inline src/Categories/Functor/Coalgebra.agda:25]: Can you make these 2 modules 1) private, and 2) qualified with `using` clauses so that they are smaller?

**Review by JacquesCarette — CHANGES_REQUESTED** [review]
Thanks!  Could you also put some 'using' clauses in your local `open`s? They actually make the resulting saved interface smaller, and are useful documentation.

**Comment by t-wissmann** [comment]
Thanks for implementing this -- it also would be useful for me, too!

By the way, there is a space missing in the PR title ('forma' -> 'form a').

**Review by JacquesCarette — APPROVED** [review]

**Comment by JacquesCarette** [comment]
Sorry to be so slow to review this. But now there's a conflict... can you resolve, and then I can merge?

**Comment by Taneb** [comment]
This was added in #362

**Pull request closed without merging**